specifyweb/frontend/js_src/lib/components/QueryBuilder/Components.tsx

```diff
@@ -173,7 +173,10 @@ export function QueryButton({
       dialogMessage={queryText.queryDeleteIncompleteDescription()}
       disabled={disabled}
       showConfirmation={showConfirmation}
-      onConfirm={handleClick}
+      onConfirm={(): void => {
+        handleClick();
+        globalThis.location.reload();
```

Comment:
also, reloading the page is an ugly hack, not a solution
this looks like an issue with state not being updated? or updated but change not propagated? that is what should be fixed. forcing the user to reload the page as a workaround for this is a terrible UX (and note not all specify instances load as fast as the do on your expensive laptop, so we need to avoid reloading needleslly)